astronomer/providers/sftp/hooks/sftp.py

```diff
@@ -81,6 +96,69 @@ async def _get_conn(self) -> asyncssh.SSHClientConnection:
                 if "private_key" in extra_options:
                     self.private_key = extra_options.get("private_key")
 
+                host_key = extra_options.get("host_key")
+                no_host_key_check = extra_options.get("no_host_key_check")
+
+                if no_host_key_check is not None:
+                    no_host_key_check = str(no_host_key_check).lower() == "true"
+                    if host_key is not None and no_host_key_check:
```

Comment:
nitpick: Should we use `if host_key and no_host_key_check`? Or is there case such check won't work

Reply:
same comment as above